bugwarrior/services/gmail.py

```diff
@@ -178,12 +179,28 @@ def get_labels(self):
 
     def get_threads(self):
         thread_service = self.gmail_api.users().threads()
+        threads = []
 
-        result = thread_service.list(
-            userId=self.config.login_name, q=self.config.query).execute()
-        return [
-            thread_service.get(userId='me', id=thread['id']).execute()
-            for thread in result.get('threads', [])]
+        pageToken = None
+
+        while len(threads) <= self.config.email_limit:
+            maxResults = min(100, max(self.config.email_limit,
+                                      self.config.email_limit - len(threads)))
+
+            result = thread_service.list(userId=self.config.login_name, q=self.config.query,
+                                         maxResults=maxResults, pageToken=pageToken).execute()
+
+            for thread in result.get('threads', []):
+                threads.append(thread_service.get(userId='me', id=thread['id']).execute())
+
+            if len(result.get('threads', [])) == 0:
+                break
+
+            pageToken = result.get('nextPageToken', None)
+            if not pageToken:
+                break
+
+        return threads[:self.config.email_limit]
```

Comment:
Isn't this slice redundant since the logic above already guarantees that threads will be fetched up to the exact `email_limit`? So couldn't this just be `return threads`? Seems like the purpose of the limit is mostly to avoid excessive http requests anyway.

Reply:
This might have been needed correction when `<=` in while, and the problem with the `min(100, max( ... )`